packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -208,34 +219,38 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
     }
   }, [isOpen]);
 
-  // Initialize filtered options to the latest value options
-  const [filteredOptions, setFilteredOptions] = React.useState(options);
-  React.useEffect(() => {
-    setFilteredOptions(options);
-    dispatch(['focus']);
-  }, [options]);
-
   // Filter options based on input value
-  const [inputValue, setInputValue] = React.useState(inputProps?.value ?? '');
+  const [inputValue, setInputValue] = React.useState<string>(
+    inputProps?.value?.toString() ?? '',
+  );
   const handleOnInput = React.useCallback(
     (event: React.ChangeEvent<HTMLInputElement>) => {
       const { value } = event.currentTarget;
       setInputValue(value);
       dispatch(['open']); // reopen when typing
-      setFilteredOptions(
-        filterFunction?.(options, value) ??
-          options.filter((option) =>
-            option.label.toLowerCase().includes(value.toLowerCase()),
-          ),
-      );
       if (focusedIndex != -1) {
         dispatch(['focus', -1]);
       }
       inputProps?.onChange?.(event);
     },
-    [filterFunction, focusedIndex, inputProps, options],
+    [focusedIndex, inputProps],
   );
 
+  // Initialize filtered options to the latest value options
+  const [filteredOptions, setFilteredOptions] = React.useState(options);
+  React.useEffect(() => {
```

Comment:
why these needed to be moved?
Also, previously it was focusing on options change.. wont break anything?

Reply:
Needed to move because I am using `inputValue` here so it must go after previous block. It is still resetting focus on input.